### ISSUE TITLE
React Class CodeMod

### ISSUE THREAD

**Issue by iammerrick** [facebook/react#4628]
So when the right hand side of a method definition is an expression it gets dropped completely by the code mod.. Example:

```javascript
function createMethod(type) {
  return function() {}
}
var Component = React.createClass({
  method: createMethod('tag')
});
```

The result is the code mod completely dropping the method
```javascript
class Component extends React.Component {}
```

**Comment by zpao** [comment]
cc @cpojer

**Comment by iammerrick** [comment]
Thanks for the mod made refactoring loads easier :+1:

**Comment by cpojer** [comment]
We should probably warn in that case or wrap it. Thank you!

**Comment by cpojer** [comment]
This should probably use ClassProperty behind a flag and throw/warn if that flag is not enabled.

**Comment by fkling** [comment]
Could also inline into the constructor.

**Comment by cpojer** [comment]
Constructor means something different than inlining it though. Most of the time I guess it is fine but the correct fix is probably

```js
MyClass.prototype.foo = fooBar()
```

**Comment by fkling** [comment]
This was primarily in response to your ClassProperty suggestion. If you can use a ClassProperty you can also put it in the constructor for better compatibility  (that's what I meant). But yeah, the correct fix is actually adding it to the prototype.

**Comment by cpojer** [comment]
I finally got around to do this. I solved this without an automated codemod but I'll see what other patterns I can find in the wild and might reconsider. I think this is usually an indication of doing something in an obscure way so I'll leave it up to the developer to fix the class before running the codemod. They can then pick whether they want to put something on the prototype, inline it in the constructor or use class properties – it would require better static analysis tools to infer whether that function call or non-simple function declaration will have side effects etc.
